$OpenBSD: patch-lib_autoconf_c_m4,v 1.1.1.1 2021/01/16 11:04:06 ajacoutot Exp $

_AC_PROG_CC_C99: fix typo (#110396)

_AC_PROG_CC_C99 was using the wrong test program.

Fixes #110396, reported anonymously.

* lib/autoconf/c.m4 (_AC_PROG_CC_C99): Use the C99 test program, not
  the C89 test program.


Index: lib/autoconf/c.m4
--- lib/autoconf/c.m4.orig
+++ lib/autoconf/c.m4
@@ -1559,7 +1559,7 @@ dnl SVR4			-Xc -D__EXTENSIONS__
 # option to output variable CC to make it so.
 AC_DEFUN([_AC_PROG_CC_C99],
 [AC_REQUIRE([_AC_C_C99_TEST_PROGRAM])]dnl
-[_AC_C_STD_TRY([c99], [ac_c_conftest_c89_program],
+[_AC_C_STD_TRY([c99], [ac_c_conftest_c99_program],
 dnl Try
 dnl GCC		-std=gnu99 (unused restrictive modes: -std=c99 -std=iso9899:1999)
 dnl IBM XL C	-qlanglvl=extc1x (V12.1; does not pass C11 test)
